Cargo.toml

```diff
@@ -58,9 +58,9 @@ hyper-rustls = { version = "0.27.1", default-features = false }
 hyper-socks2 = { version = "0.9.0", default-features = false }
 hyper-timeout = "0.5.1"
 hyper-util = "0.1.9"
-json-patch = "3"
+json-patch = "3.0.1"
```

Comment:
is this actually a tighter bound? i'm pretty sure this is ignored on the app side. you'd have to pin with `^` afaikr

Reply:
It is tighter, due to server rules - Cargo won't update it to 3.1 🙂 

I first tried with pinning it, verified that this fixed the issue, and then removed the pin to just specify the minimum required version - this still works (verified locally) and it will allow patch updates but not minor version updates.

Reply:
Looking at [CI logs](https://github.com/kube-rs/kube/actions/runs/13498319538/job/37717172658?pr=1701), it seems that build is pulling in `jsonptr v0.7.1` regardless of our preference. I guess if they use it, we will have to use the one they use anyway.

Reply:
In related news, [3.1.0 was yanked and 4.0.0 was released](https://crates.io/crates/json-patch/versions), so probably will give this a day or two, then update manually.

I see they are also re-exporting [jsonptr now](https://github.com/idubrov/json-patch/commit/e3d5c8347893d4347e9febaae50321bb3b649fd5) so we can stick to pinning one dep.